modules/azure-public-dns/data.tf

```diff
@@ -0,0 +1,3 @@
+data "azuread_group" "dnszone_contributor_group"{
```

Comment:
formatting looks wrong on this file, I would also move it to the rbac.tf file it's not useful outside of that file